src/ipc/ipc4/handler.c

```diff
@@ -463,9 +463,10 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 	struct ipc4_chain_dma cdma;
 	struct ipc *ipc = ipc_get();
 	bool delay = false;
-	int ret;
+	int ret = memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(*ipc4));
 
-	memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(*ipc4));
+	if (ret < 0)
+		return IPC4_FAILURE;
```

Comment:
@lyakh 
- commit message: ""so it makes no sense to add gradeful run-time error handling there. Instead
add assert()s to protect against possible future human error.""
- actual patch: adds runtime error handling

Hmmm.

Reply:
@kv2019i haha, the reader is invited to read between lines ;-) Corrected, thanks.